### ISSUE TITLE
Mahnung per Mail versenden

### ISSUE THREAD

**Issue by HendrikD** [HendrikD/obiblio#1]
Eine Mahnung soll anstatt per Brief auch als Mail versendet werden können.

- Haben Kunden bereits Mailadressen die hinterlegt werden können?
- Generiert openbiblio PDF-Dateien für den Briefversand? Diese könnte man dann einfach als Anhang senden
- Es braucht eine Konfigurationsmöglichkeit um den Text der Mail und den Betreff festlegen zu können
- Der zu verwendende Mailserver muss möglicherweise konfigurierbar sein (oder wir nehmen einfach sendmail() )

Mailadresse kann gepflegt werden
![image](https://user-images.githubusercontent.com/932437/176260783-997d52b3-5d32-4d18-bc52-63ebbbcced16.png)

**Comment by LeAchim59** [comment]
Soweit ich den Code richtig interpretiere und das Beispiel zeigt das auch, werden einfach Seiten im Browser erzeugt, die dann gedruckt werden.
in openBiblio 1.0 gibt es PDF Erzeugung: https://github.com/openbiblio/openbiblio-1.0
Kann man aber nur als eine mögliche Implementierungsvariante sehen, die Datenbankstruktur hat sich wohl stark geändert.

**Comment by HendrikD** [comment]
Ja meine aktuelle Idee würde auch eher in die Richtung gehen das wir eine .php Seite machen die per Cron aufgerufen wird und dann Daten aus der Datenbank lädt und die Mahnungen generiert. Wir müssten dann noch markeiren das wir eine Mahnung versendet haben damit die Person nicht täglich Mahnungen bekommt. Die Frage ist auch ob und wann wir Folgemahnungen versenden wollen und was wir machen wenn jemand die Mahnungen einfach ignoriert.

**Comment by HendrikD** [comment]
Und ich würde glaube ich auf den PDF-Anhang erst mal verzcihten und das als Textmail versenden. Wenn wir das haben können wir immer noch über einen PDF-Anahng nachdenken.

**Comment by LeAchim59** [comment]
Wir müssen das ganze Mahnverfahren noch mal im Rahmen der Ausleihbedingungen besprechen, also wann, wie oft mit welchen Konsequenzen (Sperrung des Kontos, Gebühren)
Dass eine Mahnung versendet wurde, müsste das System eigentlich wissen, aber wie genau Marcus das handhabt müsste man ihn noch mal fragen.

**Comment by HendrikD** [comment]
So wie ich das aktuell sehe merkt sich das System nicht ob eine Mahnung verschickt wurde. Es bietet einfach nur an Mahnungen für überfällige Medien zu erstellen um die dann selbst auszudrucken. Was man da ausgedruckt hat und wie man das dann trackt scheint nicht Teil des Systems zu sein. Jedenfalls finde ich da nichts zu.

**Comment by LeAchim59** [comment]
Ja, das kann gut sein. Mal Marcus fragen, wie er so damit auskommt. 
Natürlich wäre eine Verwaltung der Mahnungen prima. Zumindest den Status "gemahnt" sollte man dann einführen. Evtl. mit Mahndatum. Dazu braucht man die DB-Struktur auch nicht zu ändern. Ginge dann sogar ohne Status, wenn ein Mahndatum vorhanden ist.

**Comment by LeAchim59** [comment]
Wir haben gestern mal kurz über das Mahnverfahren gesprochen und uns auf meinen Vorschlag geeinigt:
Erinnerungsmail nach Ablauf der Leihfrist (wenn die immer verschickt wird, muss das auch nicht unbedingt in der DB vermerkt werden. Evtl. könnte man eine Logdatei oder -tabelle anlegen, so dass man eine Info hat, wieviele Erinnerungen so rausgehen.
Dann nach n Tagen (n muss noch festgelegt werden) eine Mahnung per Mail mit Mahngebühr und Sperrung des Kontos. Wenn ich mich recht erinnere, ist ein Ausleihkonto sowieso gesperrt, sobald da offene Gebühren drauf sind. Bisher rechnet das System eine Überziehungsgebühr bei Rückgabe aus.
Wenn dann immer noch nicht innerhalb eine gewissen Frist die Rückgabe erfolgte, reicht eine Email nach Ansicht der Gruppe nicht mehr. Evtl. wird man telefonieren oder einen klassischen Brief schreiben müssen, auf jeden Fall ist eine Benachrichtigung eines Ludothek-Mitarbeiters notwendig.

**Comment by HendrikD** [comment]
Ich denke beim Mahnwesen haben wir noch etwas Zeit. Es sit nicht furchtbar dringend für die Eröffnung.

1. Dauert es ja etwas bis die ersten Mahnungen nötig werden würden.
2. Kann man zumindest eine Weile das Problem auch manuell angehen.

Ist daher kein Blocker für die Eröffnung (so wie die Passwörter) aber wollen wir natürlich zeitnah haben.

**Comment by LeAchim59** [comment]
Sehe ich auch so. 
Wir haben jetzt den 1.9 für die offizielle Eröffnung vorgesehen. Die Datenbank sollten wir dann ungefähr spätestens 2 Wochen vorher für Tests, Schulungen und Eingabe der Spiele haben.

**Comment by HendrikD** [comment]
Dann müssen wir uns langsam Gedanken machen wer wann die Software auf dem Produktivsystem aufpielt.

**Comment by LeAchim59** [comment]
Dann habe ich mal die Datenbank angelegt:

Datenbank: dbs8224896
Benutzername: dbu2005690 
Passwort kommt in Email
MySQL-Version: 5.7
phpMyAdmin: https://www.strato.de/apps/phpmyadmin/strato-app/myadmin4824/dbs8224896/index.php?db=dbs8224896


Die Datenbank ist leer. PHP steht noch auf Version 7.4

**Comment by LeAchim59** [comment]
Das Mahnverfahren hat sich aus Zahlungstechnischen Gründen gestern etwas geändert.
Es bleibt bei der Erinnerung am Ende der Ausleihfrist und weiteren Erinnerungen nach jeweils x Tagen.

Die Gebühren werden aber vereinfacht: Für jede volle Woche Überziehung werden 2€ abgebucht. Dazu muss eine Liste der fälligen Beträge bis zum Stichtag abgebucht werden. Diese Liste wird 1x monatlich erzeugt. Beim Spiel muss vermerkt werden, bis wann gezahlt wird. Im Prinzip verlängert sich dadurch die Rückgabefrist, sollte aber eigenes Datenfeld sein, damit das ursprüngliche Rückgabedatum erhalten bleibt.
